### PR TITLE
ADD: Long Term Form

### PR DESCRIPTION
**Pull request by shrutiyer**

### CODE REVIEW

File: public/javascripts/ngApp.js [hunk 1]
```diff
@@ -58,7 +58,15 @@ springInitiative.config(function($stateProvider, $urlRouterProvider,
...
         controller: 'addEntryController',
-        templateUrl: 'views/content/entryForm.html'
+        templateUrl: 'views/content/dailyEntry.html'
+      }
+    }
+  })
+  .state('index.student.addLongEntry', {
+    views: {
+      'studentView@index.student': {
+        controller: 'addEntryController',
```
Comment (nmohamed) [inline public/javascripts/ngApp.js]: should make new controller instead

File: public/javascripts/studentControllers.js [hunk 2]
```diff
@@ -28,17 +28,26 @@ var studentController = function($scope,  $http, $state) {
...
 var addEntryController = function($scope, $http, $location) {
-  // $scope.minSlider = {
-  //   value: 10
-  // };
+  $scope.newEntry = {engageContent: 5, engagePeer: 5};
   $scope.submitNewEntry = function(student) {
     $http.post('api/student/newEntry/' + student._id, $scope.newEntry)
     .then(function successCallback(response) {
       $scope.entrySubmittedMsg = response.data.msg;
+      $scope.newEntry = {engageContent: 5, engagePeer: 5};
```
Comment (nmohamed) [inline public/javascripts/studentControllers.js]: why declare this twice? also, should it be null like in line 46?

File: public/views/content/dailyEntry.html [hunk 3]
```diff
@@ -82,21 +71,23 @@
       </div>
 
       <div class="form-group">
-        <label>Reading Level:</label>
-        <input type="text" ng-model="newEntry.readingLevels" placeholder="Input current reading level of student." class="form-control" />
+        <label>Teacher Feedback:</label>
+        <input type="text" ng-model="newEntry.teacherFeedback" placeholder="Input any teach feedback." class="form-control" />
```
Comment (nmohamed) [inline public/views/content/dailyEntry.html]: input type="text" ng-model="newEntry.teacherFeedback" placeholder="Input any `teacher` feedback." class="form-control"

File: routes/entryRoutes.js [hunk 4]
```diff
@@ -4,37 +4,64 @@ var routes = {};
...
     }
-    console.log("new form entry", newEntryObj)
-    console.log(studentID, date, period, attendance, warnings, behaviorText, stars, warnings, actionSteps, teacherFeedback)
+    console.log(studentID, date, period, attendance, warnings, behaviorText, stars, actionSteps)
+    res.json({newEntryObj:newEntryObj, msg: 'Entry submitted successfully!'});
+  })
+};
 
+routes.POSTnewLongEntry = function(req, res, next) {
+  // This route only handles short term
```
Comment (nmohamed) [inline routes/entryRoutes.js]: long term?

File: routes/entryRoutes.js [hunk 5]
```diff
@@ -4,37 +4,64 @@ var routes = {};
 var FormDB = require(path.join(__dirname, '../models/form'));
 
 routes.POSTnewEntry = function(req, res, next) {
```
Comment (nmohamed) [inline routes/entryRoutes.js]: lets rename this to `newDailyEntry`

**Comment by nmohamed** [comment]
:sunny: :evergreen_tree: :rose: :bouquet: :tulip:  :sunflower: :deciduous_tree: :cloud:

File: public/javascripts/studentControllers.js (hunk 2 above)
Comment (shrutiyer) [inline public/javascripts/studentControllers.js]: This line basically nulls the form and resets the sliders to this value. Adding just the null messes up with the sliders and adding null along with this line isn't doing anything new

File: routes/entryRoutes.js (hunk 4 above)
Comment (shrutiyer) [inline routes/entryRoutes.js]: Long term feedback (for ex monthly, every 9 weeks etc)

File: public/javascripts/ngApp.js (hunk 1 above)
Comment (davidabrahams) [inline public/javascripts/ngApp.js]: agreed

File: routes/entryRoutes.js [hunk 6]
```diff
@@ -4,37 +4,64 @@ var routes = {};
...
-  var period = req.body.period;
   var attendance = req.body.attendance;
   var behaviorText = req.body.behaviorText;
   var actionSteps = req.body.actionSteps;
   var schoolBehavior= req.body.schoolBehavior;
-  var readingLevels = req.body.readingLevels;
   var teacherFeedback = req.body.teacherFeedback;
   var warnings = req.body.warnings;
   var stars = req.body.stars;
+  var engageContent = parseInt(req.body.engageContent);
```
Comment (davidabrahams) [inline routes/entryRoutes.js:19]: why does this have to be parsed?

Comment (shrutiyer) [inline routes/entryRoutes.js:19]: I think I left it there from debugging. It should work without parsing. I can remove it

Comment (davidabrahams) [inline routes/entryRoutes.js:19]: :+1:

**Comment by davidabrahams** [comment]
Other than my one comment, :+1:

**Comment by BrennaManning** [comment]
:+1:  :+1:  :+1:

**Comment by davidabrahams** [comment]
merging!

**Pull request merged**